torch/_guards.py

```diff
@@ -344,6 +344,52 @@ def restore_graphstate(self, state):
         self.nn_modules = state.nn_modules
 
 
+class GlobalContextCheckpointState:
+    global_state: Dict[str, bool] = {}
+
+    def __init__(self, global_states):
+        self.global_state = global_states
+
+    """
+    Produces a delta against another GlobalContextCheckpointState.
+
+    Returns None if no delta is found, otherwise, return a set() of mismatched
+    global key names.
+    """
+
+    def diff(self, other):
+        r = set(self.global_state.keys()).difference(set(other.global_state.keys()))
+        if len(r) == 0:
+            return None
+        return r
+
+    def __eq__(self, other):
+        return self.diff(other) is None
+
+
+class GlobalContext(Checkpointable[GlobalContextCheckpointState]):
+    """
+    This keeps track of the global torch state during tracing of a function.
+    For example, torch.is_grad_enabled.
+    """
+
+    def __init__(self):
+        self.global_state: Dict[str, bool] = {}
+
+    def copy_graphstate(self):
+        return GlobalContextCheckpointState(dict(self.global_state))
+
+    def restore_graphstate(self, state):
```

Comment:
These need to be added to the copy/restore flow on TX

Reply:
Can you elaborate more? I am calling these during the copy/restore in the `output_graph`

Reply:
Oh I missed that this is in output graph. That works for me! 

Consider moving it up to Tx, it fits there better, but no strong opinions.